lms/djangoapps/discussion/rest_api/tests/test_tasks.py

```diff
@@ -230,3 +239,224 @@ def test_notification_is_send_to_cohort_ids(self, cohort_text, notification_type
+            _get_mfe_url(self.course.id, self.thread.id)
+        )
+        self.assertEqual(args_comment.app_name, 'discussion')
+
+        # check if the notification is sent to the parent response creator
+        self.assertEqual([int(user_id) for user_id in args_comment_on_response.user_ids], [self.user_2.id])
+        self.assertEqual(args_comment_on_response.notification_type, 'new_comment_on_response')
+        expected_context = {
+            'replier_name': self.user_3.username,
+            'post_title': self.thread.title,
+            'course_name': self.course.display_name,
+        }
+        self.assertDictEqual(args_comment_on_response.context, expected_context)
+        self.assertEqual(
+            args_comment_on_response.content_url,
+            _get_mfe_url(self.course.id, self.thread.id)
+        )
+        self.assertEqual(args_comment_on_response.app_name, 'discussion')
+
+    def test_no_signal_on_creators_own_thread(self):
+        """
+        Makes sure that no signal is emitted if user creates response on
+        their own thread.
+        """
+        handler = mock.Mock()
+        USER_NOTIFICATION_REQUESTED.connect(handler)
+        send_response_notifications(self.thread.id, str(self.course.id), self.user_1.id, parent_id=None)
+        self.assertEqual(handler.call_count, 0)
+
+    def test_comment_creators_own_response(self):
+        """
+        Check incase post author and response auther is same only send
+        new comment signal , with your as author_name.
+        """
+        handler = mock.Mock()
+        USER_NOTIFICATION_REQUESTED.connect(handler)
+
+        self.register_get_comment_response({
+            'id': self.thread_3.id,
+            'thread_id': self.thread.id,
+            'user_id': self.thread_3.user_id
+        })
+
+        send_response_notifications(self.thread.id, str(self.course.id), self.user_3.id, parent_id=self.thread_2.id)
+        # check if 1 call is made to the handler i.e. for the thread creator
+        self.assertEqual(handler.call_count, 1)
+
+        # check if the notification is sent to the thread creator
+        args_comment = handler.call_args_list[0][1]['notification_data']
+        self.assertEqual(args_comment.user_ids, [self.user_1.id])
+        self.assertEqual(args_comment.notification_type, 'new_comment')
+        expected_context = {
+            'replier_name': self.user_3.username,
+            'post_title': self.thread.title,
+            'author_name': 'your',
+            'course_name': self.course.display_name,
+        }
+        self.assertDictEqual(args_comment.context, expected_context)
+        self.assertEqual(
+            args_comment.content_url,
+            _get_mfe_url(self.course.id, self.thread.id)
+        )
+        self.assertEqual(args_comment.app_name, 'discussion')
+
+
+@override_waffle_flag(ENABLE_NOTIFICATIONS, active=True)
+class TestSendCommentNotification(DiscussionAPIViewTestMixin, ModuleStoreTestCase):
```

Comment:
can we add notification type specific tests here as well? right now we are just checking for the `send_notification` task!
Checking individual notification types, and their pre-conditions (conditions when they are sent and not sent) would be a more robust approach!